hibernate-core/src/main/java/org/hibernate/type/descriptor/java/BooleanJavaType.java

```diff
@@ -193,10 +193,13 @@ public String getCheckCondition(String columnName, JdbcType jdbcType, BasicValue
 						(BasicValueConverter<Boolean, ?>) converter;
 				final Object falseValue = stringConverter.toRelationalValue( false );
 				final Object trueValue = stringConverter.toRelationalValue( true );
-				String[] values = new String[] {
-						falseValue != null ? falseValue.toString() : null,
-						trueValue != null ? trueValue.toString() : null
-				};
+				final Object nullValue = stringConverter.toRelationalValue(null);
```

Comment:
We might want to surround this with a try-catch block to avoid NPEs for existing converters not handling `null`.

```suggestion
Object nullValue = null;
try {
	nullValue = stringConverter.toRelationalValue( null );
} catch (Exception e) {
	// ignored
}
```

Reply:
Sure I'll add a try catch. 
The interface is explicitly `nullable`
```java 
@Nullable R toRelationalValue(@Nullable D domainForm);
``` 
But your point is well taken.